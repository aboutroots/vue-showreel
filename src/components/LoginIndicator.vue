<template>
  <div class="login-indicator">
    <div v-if="!isAuthenticated">Anonymous user</div>
    <div v-else class="auth-wrap">
      <span class="sub"> Signed in as </span>
      <span class="username">{{ currentUser.email }}</span>
      <div class="logout-icon" @click="onLogout" @keyup.enter="onLogout">
        <ios-log-out-icon w="20" h="20" />
      </div>
    </div>
  </div>
</template>

<script lang="ts">
import { mapState, mapGetters, mapActions } from 'vuex';

import Vue from 'vue';

export default Vue.extend({
  name: 'LoginIndicator',
  computed: {
    ...mapGetters('auth', ['isAuthenticated']),
    ...mapState('auth', ['currentUser']),
  },
  methods: {
    ...mapActions('auth', ['LOGOUT_USER']),
    onLogout(): void {
      this.LOGOUT_USER();
      this.$router.push({ path: '/' });
      this.$toast.open('Signed out!');
    },
  },
});
</script>

<style lang="scss" scoped>
@import '@/assets/styles/theme.scss';

.login-indicator {
  margin-bottom: 0.5rem;
  flex: 1;
  > * {
    text-align: right;
  }
}

.auth-wrap {
  color: $gray;
  display: flex;
  justify-content: flex-end;
  flex-wrap: wrap;
  align-items: center;
  .username {
    margin: 0 0.5rem;
    color: $light;
  }
  .logout-icon {
    cursor: pointer;
  }

  .sub {
    display: none;
  }

  @include tabletUp {
    .sub {
      display: block;
    }
  }
}
</style>
