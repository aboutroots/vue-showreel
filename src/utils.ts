export const sleep = async (ms: number): Promise<void> => {
  /**
   * Sleep for given number of miliseconds
   */
  // eslint-disable-next-line no-promise-executor-return
  await new Promise((r) => setTimeout(r, ms));
};
