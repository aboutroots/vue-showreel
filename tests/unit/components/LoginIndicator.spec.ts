import LoginIndicator from '@/components/LoginIndicator.vue';
import { shallowMount } from '../utils';

describe('LoginIndicator tests suite', () => {
  test('renders the component successfully', () => {
    const wrapper = shallowMount(LoginIndicator);

    const container = wrapper.find('.LoginIndicator');

    expect(container.exists()).toBeTruthy();
  });

  test('matches snapshot', () => {
    const wrapper = shallowMount(LoginIndicator);
    expect(wrapper.element).toMatchSnapshot();
  });
});
