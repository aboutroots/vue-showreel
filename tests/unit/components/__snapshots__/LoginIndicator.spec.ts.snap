// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginIndicator tests suite matches snapshot 1`] = `
<div
  class="LoginIndicator login-indicator"
>
  <div>
    Anonymous user
  </div>
</div>
`;
